### PR TITLE
feat: PATRON2020-234 add endpoint for notifications, add schema for endpoint

### PR DESCRIPTION
**Pull request by jwieckowski**
provides endpoints for getting all notifications and deleting seen ones, defined endpoint stands for app/v1/notifications. Db interaction is also defined, it allows to keep only 100 latest notifications in the app

### CODE REVIEW

File: frontend/src/data/reducers/notification.js [hunk 1]
```diff
@@ -40,12 +43,30 @@ export default function notification (state = initialState, action = {}) {
         isDrawerOpen: false
       }
     case actionTypes.NOTIFICATIONS_CHECK:
-      return state
-
+      return {
+        ...state,
+        notificationContent: state.notifications.filter(n => n.id === action.id),
```
Comment (MartaJaszewska) [inline frontend/src/data/reducers/notification.js:48]: Shouldn't it be in some saga?

File: frontend/src/data/sagas/notification/notificationSaga.spec.js [hunk 2]
```diff
@@ -136,19 +140,45 @@ describe('checkNotificationsSaga', () => {
...
-    sensorId: 55
-  }]
-  const checked = [{
+  const mockNotifications = [
+    {
+      id: 1,
+      timestamp: 1500000100,
+      type: 'RFIDSensor',
+      sensorId: 55,
+      isChecked: false
```
Comment (MartaJaszewska) [inline frontend/src/data/sagas/notification/notificationSaga.spec.js:149]: We don't add 'isChecked: false' in check saga.

File: src/plugins/db/actions/index.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+const notifications = require('./dbNotificationsActions')
+
+module.exports = fp(function (fastify, options, next) {
+  fastify.decorate('db', {
+    getAllSensors: sensors.getSensors,
+    postOneSensor: sensors.postSensor,
+    removeOneSensor: sensors.removeSensor,
+    getAllNotifications: notifications.getNotifications,
+    postOneNotification: notifications.postNotification,
+    updateOneNotification: notifications.updateNotification
```
Comment (karlosos) [inline src/plugins/db/actions/index.js:12]: Why there's a change of method names?

Comment (jwieckowski) [reply src/plugins/db/actions/index.js:12]: You're right, names should be the same, I will change it

File: src/plugins/db/actions/dbSensorsActions.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
-const fp = require('fastify-plugin')
+// const fp = require('fastify-plugin')
```
Comment (karlosos) [inline src/plugins/db/actions/dbSensorsActions.js:1]: Why did you comment this line out? Shouldn't this line be deleted?

File: tests-server/notifications.spec.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        method: 'GET',
+        url: '/api/v1/notifications'
+      })
+
+      expect(instance.config.GATEWAY_URL).toBe('https://patronage20-concept-master.herokuapp.com')
+      expect(result.statusCode).toBe(200)
+      expect(typeof (JSON.parse(result.payload))).toBe('object')
+    })
+
+    test('sshould fastify.config.GATEWAY_URL be set and status code should be 400 when no notifications in db', async function () {
```
Comment (karlosos) [inline tests-server/notifications.spec.js:50]: Typo in "sshould".

**Pull request merged**